### PR TITLE
Update Repo to Node v8 LTS, add typescript and reafctor code to use new lang features

### PR DESCRIPTION
**Pull request by slaby93**
Hi Everyone!
I saw that package is using old Node v4 while it's already deprecated. I updated it to Node v8 which has support up to 2019.
While doing this I also updated repo to use typescript at it makes using and writing lib much easier.
Also added logger file to make logging to user more unified. Everything is tested.
Please, let me know WDYT?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,11 +2,16 @@
   "name": "heroku-self-ping",
   "version": "1.1.5",
   "author": "Matthieu Bacconnier <matthieu@anyfetch.com>",
+  "description": "Node.js script for keeping your Heroku application alive. Avoid the one-hour time limit on your dyno.",
```
Comment (Neamar) [inline package.json]: I think this is now 30 minutes :(

**Comment by Neamar** [comment]
Hello!

First off, thanks for your time contributing this :)

I'm not a TypeScript user myself -- I don't have anything against it, but I'd like the module to keep working with "standard" JS. From what I can see, if I were to install the module now in plain JS, it would fail, right?
That's something I'd like to avoid, I know Typescript is growing but there is still a lot of plain JS users :)

Otherwise, it looks good, no complaints from me!

You might even be able to bump Node to v10, which is the new LTS, I don't expect any issues!

**Comment by slaby93** [comment]
Hi @Neamar 
So you can install whatever module you want :) Trick is that if you want to use modules that doesn't have types, just use require over import. Rest is exactly the same. I saw that lots of libs are written in typescript because thanks to it, you can export types, which prevents lots of bugs (even if user doesnt use typescript, IDEs still can prompt lib user with correct types).

**Comment by slaby93** [comment]
You can merge it or not, but I would just suggest you to play a little bit with it :) I was sceptic about TS myself for a long time, but when I developed a couple of small libs in TS, I saw that the most of errors were discovered during writing code ( you have an option that TS won't compile files if he finds any error in it )

**Comment by Neamar** [comment]
I might do that at some point :)

But in the meantime, can't we add the compiled js file to the repo too to make sure plain JS users can require() it normally?

**Comment by slaby93** [comment]
@Neamar you should never do it. Source files are suppose to be readable while transpiled files are for users to require, thus they are minified and optimized for computers. In package.json you've a JS task called `build` which generates `dist/*` folder with normal `.js` files. When you publish your project, there is a couple hooks that transpiles files when you publish. Reason to do that is because people oftem makes mistakes. Keeping `dist/*` folder with transpiled files in repo might contain older version then source code.

Again, this means that when you publish this project on npm, those files will be generated, so no point to keeping them.

You can read how it works here: https://itnext.io/step-by-step-building-and-publishing-an-npm-typescript-package-44fe7164964c

**Comment by Neamar** [comment]
Hello!

Sorry for the long delay, I was finally able to test and it looks good!

One small issue with Mocha not finishing automatically, at least on my computer.

I've added the `--exit` flag and it's now perfect!

I'll publish to npm as a major, since we're dropping support for old nodejs versions (string interpolation).

**Comment by Neamar** [comment]
See ff883b7 for exit flag, 

Package published to npm as v2.0.0

Thanks for your help!

**Pull request merged**